src/main/java/com/cosmian/jna/findex/ffi/Progress.java

```diff
@@ -29,13 +27,12 @@ public boolean apply(Pointer searchResultsPointer,
         //
         // Deserialize search results
         //
-        List<IndexedValue> indexedValues =
-            Leb128Reader.deserializeCollection(IndexedValue.class, serializedSearchResults);
+        ProgressResults results = new Leb128Reader(serializedSearchResults).readObject(ProgressResults.class);
 
         //
         // Convert to Indexed Values list
         //
-        return this.progress.notify(indexedValues);
+        return this.progress.notify(results.getResults());
```

Comment:
Return `ProgressResults` and not an `HashMap` it's better to add helpers functions afterward